src/FakeClient/FakeCaptureEventProcessor.h

```diff
@@ -10,43 +10,79 @@
 
 namespace orbit_fake_client {
 
-// This implementation of CaptureEventProcessor simply discards all the events it receives, but it
-// keeps track of their number and total size and then outputs these statistics to file.
+// This implementation of CaptureEventProcessor mostly discard all events it receives, except for:
+// - keeping track of their number and total size, and writing these statistics to file;
+// - keeping track of the calls to the frame boundary function, and possibly writing the average
+//   frame time to file;
 class FakeCaptureEventProcessor : public orbit_capture_client::CaptureEventProcessor {
  public:
   void ProcessEvent(const orbit_grpc_protos::ClientCaptureEvent& event) override {
     ++event_count_;
     byte_count_ += event.ByteSizeLong();
+
+    if (event.has_function_call()) {
+      const orbit_grpc_protos::FunctionCall& function_call = event.function_call();
+      if (function_call.function_id() == kFrameBoundaryFunctionId) {
+        uint64_t start_timestamp_ns =
```

Comment:
I was a bit confused here. I expected just summing up all the `function_call.duration_ns()`, count the frames and do a division below. But this should work as well.

Reply:
If I sum the durations of the calls I get the total time spent in `vkQueuePresentKHR`, while I want the sum of the times between the calls. Or did I understand incorrectly?

Reply:
Oh, right, I was fooled by the word "duration". As you said that is the duration of the call `vkQueuePresentKHR` not the frame.